File: app/src/pro/java/com/clockmods/pro/alarm/AlarmStore.java
package com.clockmods.pro.alarm;

import android.content.Context;
import android.content.SharedPreferences;

public final class AlarmStore {
    private final SharedPreferences preferences;

    public AlarmStore(Context context) {
        preferences = context.getSharedPreferences("pro_alarm", Context.MODE_PRIVATE);
    }

    public int hour() { return preferences.getInt("hour", 7); }
    public int minute() { return preferences.getInt("minute", 30); }
    public boolean enabled() { return preferences.getBoolean("enabled", false); }

    public void save(int hour, int minute, boolean enabled) {
        preferences.edit().putInt("hour", hour).putInt("minute", minute)
                .putBoolean("enabled", enabled).apply();
    }

    public void setEnabled(boolean enabled) {
        preferences.edit().putBoolean("enabled", enabled).apply();
    }
}

File: app/src/ultimate/java/com/clockmods/pro/alarm/AlarmStore.kt
package com.clockmods.pro.alarm

import android.content.Context
import android.content.SharedPreferences

class AlarmStore(context: Context) {
    private val preferences: SharedPreferences =
        context.getSharedPreferences("pro_alarm", Context.MODE_PRIVATE)

    fun hour(): Int = preferences.getInt("hour", 7)
    fun minute(): Int = preferences.getInt("minute", 30)
    fun enabled(): Boolean = preferences.getBoolean("enabled", false)

    /**
     * True while an alarm is actively ringing.  Persisted so the ringing surface can be rebuilt
     * after the process is killed (low memory) or the device reboots mid-ring, which would
     * otherwise leave the user with a ringing alarm and no way to stop it.
     */
    fun isRinging(): Boolean = preferences.getBoolean("ringing", false)

    fun setRinging(ringing: Boolean) {
        preferences.edit().putBoolean("ringing", ringing).apply()
    }

    fun save(hour: Int, minute: Int, enabled: Boolean) {
        preferences.edit()
            .putInt("hour", hour)
            .putInt("minute", minute)
            .putBoolean("enabled", enabled)
            .apply()
    }

    fun setEnabled(enabled: Boolean) {
        preferences.edit().putBoolean("enabled", enabled).apply()
    }
}
